cf_xarray/helpers.py

```diff
@@ -211,11 +211,54 @@ def _bounds_helper(values, n_core_dims, nbounds, order):
             vertex_vals = np.block([[bot_left, bot_right], [top_left, top_right]])
     elif n_core_dims == 1 and nbounds == 2:
         # Middle points case (1D lat/lon)
-        vertex_vals = np.concatenate((values[..., :, 0], values[..., -1:, 1]), axis=-1)
+        vertex_vals = _get_ordered_vertices(values)
 
     return vertex_vals
 
 
+def _get_ordered_vertices(bounds: xr.DataArray) -> np.ndarray:
+    """Extracts a sorted 1D array of unique vertex values from a bounds DataArray.
+
+    This function takes a DataArray (or array-like) containing bounds information,
+    typically as pairs of values along the last dimension. It flattens the
+    bounds into pairs, extracts all unique vertex values, and returns them in
+    sorted order. The sorting order (ascending or descending) is determined by
+    inspecting the direction of the first non-equal bounds pair.
+
+    Parameters
+    ----------
+    bounds : xr.DataArray
+        A DataArray containing bounds information, typically with shape (..., 2).
+
+    Returns
+    -------
+    np.ndarray
+        A 1D NumPy array of sorted unique vertex values extracted from the
+        bounds.
+    """
+    # Convert to array if needed
+    arr = bounds.values if isinstance(bounds, xr.DataArray) else bounds
+    arr = np.asarray(arr)
+
+    # Flatten to (N, 2) pairs and get all unique values.
+    pairs = arr.reshape(-1, 2)
```

Comment:
Similarly here, why a reshape ?

A "bounds" array for this case is assumed to be of shape "(..., N, 2)", so that the corresponding "vertices" array is of shape "(..., N + 1)". Usually, There no other dimensions (bounds is (N, 2) and vertices is (N + 1,) ), but I don't think it's up to this function to check or handle this. I think it should preserve all other dimensions, only modifying the last two. That's what the  `core_dim` argument of `bounds_to_vertices` is about.

Reply:
My mistake, didn't analyze the implementation close enough and failed to realize these requirements.

I pushed fixes in [`30af8cc` (#579)](https://github.com/xarray-contrib/cf-xarray/pull/579/commits/30af8cc47e835b5377767f93967583230a09aa01) to make sure it only operates on the last two dims while preserving all other dims.